index.js

```diff
@@ -1,6 +1,20 @@
 'use strict';
 const os = require('os');
-const home = os.homedir();
+const isWindows = require('is-windows');
+
+const home = (() => {
+	const homeDir = os.homedir();
+	if (isWindows()) {
+		// Return the home directory with the user name part in 8.3 format
+		// (https://goo.gl/FxW2CB) if needed. This is a workaround for
+		// https://github.com/nodejs/node/issues/17586.
+		const tmpDir = os.tmpdir();
+		const userDir = homeDir.replace(/([^\\])+$/g, '');
+		const user83 = tmpDir.slice(userDir.length).replace(/\\.*$/, '');
```

Comment:
The `tmp` directory falls under the user’s home directory. This line extracts just the user part in the 8.3 format.

Reply:
So this will only work if the `TMP` dir is under the user dir. I have a (extremely strange) setup where:
```cmd
TEMP=c:\temp\usr
USERPROFILE=d:\refael
```
so unfortunately those two dirs don't have anything in common.

Reply:
Indeed, this patch only works in the case where environment variables are not overwritten. Which is better than not working at all, but worse than working in all cases 😅